### PR TITLE
feat: Add schema introspection

### PR DESCRIPTION
**Pull request by mykola-vrmchk**

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2418** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2418" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2418?utm_source=stack-comment).

File: packages/core/echo/echo-schema/src/schema.ts [hunk 1]
```diff
@@ -7,11 +7,93 @@ import * as pb from 'protobufjs';
...
+      kind: 'array';
+      elementType: EchoType;
+    };
+
 export type EchoSchemaField = {
   name: string;
+  type: EchoType;
+};
+
+const getFields = (type: pb.Type, typeStack: string[] = []): EchoSchemaField[] => {
```
Comment (dmaretskyi) [inline packages/core/echo/echo-schema/src/schema.ts:35]: typeStack?

File: packages/core/echo/echo-schema/src/schema.ts [hunk 2]
```diff
@@ -7,11 +7,93 @@ import * as pb from 'protobufjs';
...
+  type.fieldsArray.forEach((field) => field.resolve());
+
+  return type.fieldsArray.map((field) => {
+    const getComplexType = (type: pb.Type): EchoType => {
+      if (type.options && type.options['(object)']) {
+        return { kind: 'ref', objectType: type.fullName.replace('.', '') };
```
Comment (dmaretskyi) [inline packages/core/echo/echo-schema/src/schema.ts:41]:
```suggestion
        return { kind: 'ref', objectType: type.fullName.slice(1) };
```

File: packages/core/echo/echo-schema/src/schema.ts [hunk 3]
```diff
@@ -7,11 +7,93 @@ import * as pb from 'protobufjs';
...
+      } else {
+        return { kind: 'record', objectType: type.fullName.replace('.', '') };
```
Comment (dmaretskyi) [inline packages/core/echo/echo-schema/src/schema.ts:43]: .slice(1)

File: packages/core/echo/echo-schema/src/schema.ts [hunk 4]
```diff
@@ -7,11 +7,93 @@ import * as pb from 'protobufjs';
...
+    const getEchoType = (field: pb.Field): EchoType => {
+      if (field.resolvedType instanceof pb.Type) {
+        if (field.repeated) {
+          return { kind: 'array', elementType: getComplexType(field.resolvedType) };
+        } else {
+          return getComplexType(field.resolvedType);
+        }
+      } else if (field.resolvedType instanceof pb.Enum) {
+        // TODO(mykola): Add enums support.
+        throw new Error('Enums are not supported yet.');
```
Comment (dmaretskyi) [inline packages/core/echo/echo-schema/src/schema.ts:82]: add { kind: 'enum' } as a field value

**Review by dmaretskyi — APPROVED** [review]

**Pull request merged**